### ISSUE TITLE
Chinese translation

### ISSUE THREAD

**Issue by Changaco** [liberapay/salon#127]
Someone has requested through Weblate that a translation file be added for Chinese. I'm not sure how usable Liberapay is in China, but given that we already have Korean and Indonesian I see no reason to reject a request for Chinese.

Other opinions?

**Comment by revi** [comment]
Outside the mainland China, there are Taiwan, Hong Kong (where its local gov has more freedom than mainland), Macau (probably same as Hong Kong) as a country with Chinese as an official language, and of course there are Chinese who lives outside China (e.g. United States, France, etc...). Why not?

CF: [Chinese language @ en.wikipedia.org](https://en.wikipedia.org/wiki/Chinese_language)

**Comment by Changaco** [comment]
`zh.po` file created in <https://github.com/liberapay/liberapay.com/commit/0209f94dd85fda59d2de8251cf9a72e45e8a38fb>.

**Comment by aggsol** [comment]
You might need a second one for simplified.

Hong Kong, Macau ->	Cantonese/Traditional
People’s Republic of China ->	Mandarin/Simplified
Singapore ->	Mandarin/Simplified
Taiwan ->	Mandarin/Traditional
United States ->	Cantonese or Mandarin/Traditional

**Comment by jxtsai** [comment]
hi folks, 

I am the person requesting for Chinese version. 
Basically most Mandarin speaker can "read" the characters either in traditional or simplified. It's rather less easy to "write" the other  one which is not adopted in her/his place. 

I am translating the Traditional Chinese and might find someone to help the reviewing. Hopefully others may be able to work  the translation of  simplified Chinese later based on this effort. This would be  efficient and energy save though.

**Comment by Changaco** [comment]
@tuxayo Why have you been tagging so many of the Chinese translations as needing review lately?

**Comment by tuxayo** [comment]
Because almost all of them are **marked as translated** when they are not.

Case 1: Empty string
https://hosted.weblate.org/translate/liberapay/core/zh/?type=translated&offset=99#history

Case 2: Incomplete translation
https://hosted.weblate.org/translate/liberapay/core/zh/?type=translated&offset=110
https://hosted.weblate.org/translate/liberapay/core/zh/?type=translated&offset=116
https://hosted.weblate.org/translate/liberapay/core/zh/?type=translated&offset=122

It's a good surprise to see there is an issue to track the Chinese translation. Sorry for not having searched or openned and issue before beginning to mass-invalidate the strings.

Did I misunderstood the situation and flagged when I shouldn't have?

**Comment by Changaco** [comment]
The first one is not empty. I can't read Chinese at all but I can copy-paste into Google Translate, and it tells me that the 4 translations you linked to are okay (although the translation for "Donate" is reverse-translated to "Damage" or "Loss of money", which is unfortunate).

**Comment by tuxayo** [comment]
Okay, I have a terrible font issue. \*facepalm*
I actually see the field as empty. (except when there are non mandarin chars)
But when I copy paste it's content to the GitHub issue field the font, for example "管理者否決權開啟*"
Sorry for being mistakenly vandalizing the trads....
I'll gradually unflag them in the next few days/two weeks.

**Comment by tuxayo** [comment]
Even on the site there is this issue.

### Before switching to Chinese:
(note that the name of the language is visible)
![screenshot_2017-10-13_18-18-42](https://user-images.githubusercontent.com/2678215/31555886-05f81f56-b043-11e7-92c6-eb14950efff7.png)

### After switching to Chinese:
(note that the name of the language not visible anymore!)
![screenshot_2017-10-13_18-19-02](https://user-images.githubusercontent.com/2678215/31555885-05c16ad8-b043-11e7-9fe9-6fc25f79c239.png)

The font is the same so I don't get why the chars are not visible.

**Comment by Changaco** [comment]
> I'll gradually unflag them in the next few days/two weeks.

I think it's better if I remove them all at once.

**Comment by Changaco** [comment]
Done in <https://github.com/liberapay/liberapay.com/commit/24e29a91457bcc9c94d9bbc6c01160221cd1df55>.

**Comment by tuxayo** [comment]
Big thanks for that. I hope it was easy to find only the ones I flagged. So no legitimate instances were lost.

Thanks again for fixing my mess. I hope to be more cautious in the future to no begin such things without asking before if it's correct.

**Comment by austinhuang0131** [comment]
@Changaco @tuxayo FYI I have finished ~~20%~~ 30% more translations and increased the translated portion to 99.4%. Feel free to keep the work up.

**Comment by tuxayo** [comment]
@austinhuang0131 That's outstanding!

**Comment by austinhuang0131** [comment]
@tuxayo Chinese is officially done.

Please apply this translation ASAP, because there are some crucial fixes.

**Comment by austinhuang0131** [comment]
alright i finished the translation again

**Comment by jxtsai** [comment]
Hi,

I can't find the strings I translated on weblate site  and it seems only Simplfied Chinese (zh-Hans) left there.  The liberapay website still keep Tranditional Chinese version (zh-Hant) now so I wonder it's going to have simplifed Chinese version soon ? 
I would like to continue the untranslated work. Can we have   Chinese (Tranditional) zh-Hant back in weblate platform ?  

thanks in advanced

**Comment by austinhuang0131** [comment]
AFAIK in Weblate it's sort of a mix of both.

**Comment by jxtsai** [comment]
> 
> 
> AFAIK in Weblate it's sort of a mix of both.

Yes, I think we should resort it and have two Chinese languge pack.

**Comment by Changaco** [comment]
I've modified the translation file so that it appears as "Chinese (Traditional)" [in Weblate](https://hosted.weblate.org/projects/liberapay/core/).

**Comment by Changaco** [comment]
As for adding a simplified Chinese translation, I think it would be a lot of work for a small gain, unless it's automated with [OpenCC](https://github.com/BYVoid/OpenCC) or another similar tool.

If and when we add `zh_Hans` we should also add an extra fallback to and from `zh_Hant` in order to avoid showing English when a translation is missing in one of the Chinese variants but is available in the other.

Relevant issue: <https://github.com/liberapay/liberapay.com/issues/1073>.

**Comment by s8321414** [comment]
Well, if no one want to translate Simplified Chinese, OpenCC can just work, but if someone want to do it, then translation from OpenCC will be am issue.
Because some words in Simplified Chinese are not all same as Traditional Chinese. (Even all Traditional Chinese variants are not same, like Taiwan and Hong Kong, Simplified Chinese variants are the same way.)

**Comment by Changaco** [comment]
I don't see why it would be a problem, human translators would be able to review the translations and adjust them if necessary.

**Comment by LaoshuBaby** [comment]
> Because some words in Simplified Chinese are not all same as Traditional Chinese. (Even all Traditional Chinese variants are not same, like Taiwan and Hong Kong, Simplified Chinese variants are the same way.)

Yes, I'm confused why Simplified Chinese are not supported. Only Traditional Chinese can translate Librepay core.

## current translation status

* https://hosted.weblate.org/languages/zh_Hant/liberapay/

![图片](https://user-images.githubusercontent.com/42690037/152670633-97c03bac-d062-45c1-ab33-42eaa60b1b3d.png)

* https://hosted.weblate.org/languages/zh_Hans/liberapay/

![图片](https://user-images.githubusercontent.com/42690037/152670657-8cb2a596-05ff-43ec-95b9-40f2c2857920.png)

## OpenCC or not

I agree with s8321414's opinion, we can use OpenCC for auto translate when we don't have any translator. After we have a certain amount of Simplified Chinese users, community can also do manual review of these texts.

Even if you choose to start over from zero, a language option that full of fallbacks with percentage of progress showed will attract more translators.

## fallback

There are 2 kind of fallback we can choose:
1. `(zh-Hans <-> zh-Hant) -> en`, only if both S and T Chinese are not translated, then use English instead. I prefer this one because for Chinese users, it is possible to read another variant roughly (but not completely). It will make better use of the existing translation than fallback directly to English.
2. `zh-Hans -> en` + `zh-Hant -> en`, this will ignore the similarity and let those 2 varient fallback to English.
